cub/cub/device/dispatch/dispatch_scan.cuh

```diff
@@ -165,6 +165,8 @@ template <
   typename KernelLauncherFactory = detail::TripleChevronFactory>
 struct DispatchScan
 {
+  static_assert(::cuda::std::is_unsigned_v<OffsetT>, "DispatchScan only supports unsigned offset types");
```

Comment:
Should we consider even limiting to at least 4-byte wide offset types? I think we don't currently test for shorter offset types.

Reply:
Then let's also limit it to at least four-byte wide offset types

Reply:
And so it shall be